lib/modules/manager/poetry/schema.ts

```diff
@@ -35,39 +36,45 @@ const PoetryGitDependency = z
     git: z.string(),
     tag: z.string().optional().catch(undefined),
     version: z.string().optional().catch(undefined),
+    branch: z.string().optional().catch(undefined),
+    rev: z.string().optional().catch(undefined),
   })
-  .transform(({ git, tag, version }): PackageDependency => {
-    if (!tag) {
-      const res: PackageDependency = {
+  .transform(({ git, tag, version, branch, rev }): PackageDependency => {
+    if (tag) {
+      const parsedUrl = parseGitUrl(git);
+      if (parsedUrl.source === 'github.com') {
+        const { owner, name } = parsedUrl;
+        const repo = `${owner}/${name}`;
+        return {
+          datasource: GithubTagsDatasource.id,
+          currentValue: tag,
+          packageName: repo,
+        };
+      } else {
+        return {
+          datasource: GitRefsDatasource.id,
+          currentValue: tag,
+          packageName: git,
+          skipReason: 'git-dependency',
+        };
+      }
+    } else if (rev) {
+      return {
```

Comment:
```suggestion
    }
    
      if (rev) {
      return {
```
The else is unnecessary